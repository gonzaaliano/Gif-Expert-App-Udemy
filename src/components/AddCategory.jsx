import { useState } from "react";
import PropTypes from "proptypes";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        console.log(event.target.value)
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // setCategories(categories => [inputValue, ...categories]);
        setInputValue('');
        onNewCategory(inputValue.trim());
    }


    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label='form'>
            <input
                type="text"
                placeholder="Search GIFs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}