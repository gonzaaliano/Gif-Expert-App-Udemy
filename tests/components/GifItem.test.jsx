const { render, screen } = require("@testing-library/react");
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en componente <GifItem />', () => {

    const title = 'Seinfeld';
    const url = 'https://seinfeld.com/jerry.jpg';

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    })

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // console.log(screen.getByRole('img').src).toBe(url);
        // console.log(screen.getByRole('img').alt).toBe(title);
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })

    test('Debe mostrar el titulo en el componente', () => {

        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
    })
})