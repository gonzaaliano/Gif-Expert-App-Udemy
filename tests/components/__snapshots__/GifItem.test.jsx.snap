// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Seinfeld"
      src="https://seinfeld.com/jerry.jpg"
    />
    <p>
      Seinfeld
    </p>
  </div>
</div>
`;
